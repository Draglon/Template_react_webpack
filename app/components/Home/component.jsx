import React from 'react';
import PropTypes from 'prop-types';

import LoginForm from './LoginForm';
import Dashboard from './Dashboard';

const HomeComponent = ({ isLogged }) => <>{!isLogged ? <LoginForm /> : <Dashboard />}</>;

HomeComponent.defaultProps = {
  isLogged: '',
};

HomeComponent.propTypes = {
  isLogged: PropTypes.string,
};

export default HomeComponent;
