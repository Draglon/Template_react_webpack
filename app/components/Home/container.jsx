import { connect } from 'react-redux';
import { getSessionId } from '../../store/theMovieDB/login/selectors';

import HomeContainer from './component';

const mapStateToProps = state => ({
  sessionId: getSessionId(state),
});

export default connect(mapStateToProps)(HomeContainer);
