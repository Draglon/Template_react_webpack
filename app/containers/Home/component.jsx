import React from 'react';

import LoginForm from '../LoginForm';
import Dashboard from '../Dashboard';

const HomeComponent = ({ isLogged }) => <>{!isLogged ? <LoginForm /> : <Dashboard />}</>;

export default HomeComponent;
