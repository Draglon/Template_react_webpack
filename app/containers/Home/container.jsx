import { connect } from 'react-redux';
import { getSessionId } from '../../store/theMovieDB/login/selectors';

import HomeContainer from './component';

const mapStateToProps = state => ({
  isLogged: getSessionId(state),
});

export default connect(
  mapStateToProps,
  null,
)(HomeContainer);
