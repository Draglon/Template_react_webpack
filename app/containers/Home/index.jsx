import React, { Component } from 'react';
import { connect } from 'react-redux';

import { MOVIE_REQUEST } from '../../store/constants/movie.constants';

class Home extends Component {
  componentDidMount() {
    const { setMovie } = this.props;
    setMovie(3);
  }

  render() {
    return (
      <section className="section">
        <div className="wrapper">
          <h2 className="section__title">Home page</h2>
        </div>
      </section>
    );
  }
}

export default connect(
  null,
  dispatch => ({
    setMovie(payload) {
      dispatch({ type: MOVIE_REQUEST, payload });
    },
  }),
)(Home);
