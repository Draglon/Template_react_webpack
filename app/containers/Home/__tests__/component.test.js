import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Home from '../component';

describe('<NotFound /> container', () => {
  const component = shallow(<Home />);

  // it('renders without crashing', () => {
  //   expect(component.find('h2').text()).toEqual('Page not found');
  //   expect(component.find('h1').text()).toEqual('404');
  // });

  it('render property', () => {
    // expect(renderer.create(<Home />).toJSON()).toMatchSnapshot();
    expect(component).toMatchSnapshot();
  });
});
