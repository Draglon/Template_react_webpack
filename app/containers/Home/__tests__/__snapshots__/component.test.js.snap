// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> container render property 1`] = `ShallowWrapper {}`;
